src/app/public/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts

```diff
@@ -1,63 +1,79 @@
+  SkyProgressIndicatorItemStatus
+} from '../types';
 
 @Component({
   selector: 'sky-progress-indicator-item',
   templateUrl: './progress-indicator-item.component.html',
-  styleUrls: ['./progress-indicator-item.component.scss']
+  styleUrls: ['./progress-indicator-item.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class SkyProgressIndicatorItemComponent {
-  public itemNumber: number;
-  public isActive = false;
-  public isComplete = false;
-  public isLastItem = false;
-  public displayMode = SkyProgressIndicatorDisplayMode.Vertical;
-  public isPassive = false;
-  public isNextToInactive = true;
 
   @Input()
   public title: string;
 
-  public get displayTitle(): string {
-    if (this.isPassive) {
-      return this.isActive || this.isComplete ? this.title : '';
-    } else {
-      return this.itemNumber + ' - ' + this.title;
-    }
+  public get status(): SkyProgressIndicatorItemStatus {
+    return this._status;
   }
 
-  public get isLineDisplayed(): boolean {
-    return !(this.isHorizontal || this.isPassive) || this.isPassive && !this.isLastItem || this.isActive;
+  public set status(value: SkyProgressIndicatorItemStatus) {
+    if (value === this._status) {
+      return;
+    }
+
+    this._status = value;
+    this.changeDetector.markForCheck();
   }
 
-  public get headingClassList(): {[className: string]: boolean} {
-    return {
-      'sky-text-success': this.isComplete && !this.isActive && !this.isHorizontal,
-      'sky-progress-indicator-item-passive': this.isPassive,
-      'sky-progress-indicator-item-step': !this.isHorizontal && (!this.isComplete || this.isActive),
-      'sky-progress-indicator-item-step-active': this.isActive,
-      'sky-progress-indicator-item-step-complete': this.isComplete && !this.isActive && !this.isHorizontal
-    };
+  public get statusName(): string {
+    let name: string;
+
+    /* tslint:disable-next-line:switch-default */
+    switch (this.status) {
+      case SkyProgressIndicatorItemStatus.Active:
+      name = 'active';
+      break;
+
+      case SkyProgressIndicatorItemStatus.Complete:
+      name = 'complete';
+      break;
+
+      case SkyProgressIndicatorItemStatus.Incomplete:
+      name = 'incomplete';
+      break;
+
+      case SkyProgressIndicatorItemStatus.Pending:
+      name = 'pending';
+      break;
+    }
+
+    return name;
   }
 
-  public get lineClassList(): {[className: string]: boolean} {
-    return {
-      'sky-deemphasized': this.isPassive,
-      'sky-progress-indicator-item-passive': this.isPassive,
-      'sky-progress-indicator-item-horizontal': this.isHorizontal,
-      'sky-progress-indicator-item-last': this.isLastItem,
-      'sky-progress-indicator-item-line-inactive': !this.isActive && !this.isComplete,
-      'sky-progress-indicator-item-line-next-inactive': this.isNextToInactive
-    };
+  public isVisible = false;
+  public showStatusMarker = true;
+  public showTitle = true;
+  public titlePrefix: string;
```

Comment:
Are we doing that for all classes/interfaces, etc.? I thought we were only doing that for interfaces.

Reply:
Hrm... I'm not sure. Maybe we never landed on a firm decision. Don't worry about it for now :)

Reply:
Should these properties all be on their own line?